src/table/types.ts

```diff
@@ -0,0 +1,218 @@
+import { TableCellProps } from '@mui/material';
+import { stardust } from '@nebula.js/stardust';
+import {
+  Announce,
+  Cell,
+  ChangeSortOrder,
+  Column,
+  ExtendedSelectionAPI,
+  ExtendedTheme,
+  ExtendedTranslator,
+  PageInfo,
+  SetPageInfo,
+  TableData,
+  TableLayout,
+  TotalsPosition,
+} from '../types';
+import { TSelectionActions } from './utils/selections-utils';
+
+export interface SelectionState {
+  rows: Record<string, number>;
+  colIdx: number;
+  api: ExtendedSelectionAPI;
+  isSelectMultiValues: boolean;
+}
+
+export interface ActionPayload {
+  cell: Cell;
+  announce: Announce;
+  evt: React.KeyboardEvent | React.MouseEvent;
+}
+
+export interface ContextValue {
+  headRowHeight: number;
+  setHeadRowHeight: React.Dispatch<React.SetStateAction<number>>;
+  focusedCellCoord: [number, number];
+  setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>;
+  selectionState: SelectionState;
+  selectionDispatch: React.Dispatch<TSelectionActions> | jest.Mock<any, any>;
+}
+
+export interface GeneratedStyling {
+  padding?: string;
+  borderStyle: string;
+  borderColor: string;
+  fontFamily?: string;
+  color?: string;
+  fontSize?: string; // following the theme format so this should always be a string
+  cursor?: string;
+  borderWidth?: string;
+  backgroundColor?: string;
+  sortLabelColor?: string;
+  hoverBackgroundColor?: string;
+  hoverFontColor?: string;
+}
+
+export interface CellStyle {
+  backgroundColor: string | undefined; // This is always set but could be undefined in the theme
```

Comment:
Does it makes sense of making it optional?

Reply:
well I don't think so. we always set it and sometimes it undefined